js/components/resultsFormPC.js

```diff
@@ -0,0 +1,681 @@
+                            name={`${disaggregationData["652"].disaggregation_type}-${disaggregationData["652"].labels[1]}`}
+                            id={`id_${disaggregationData["652"].disaggregation_type}-${disaggregationData["652"].labels[1]}`}
+                            value={disaggregationData["652"].labels[1].value || ""}
+                            onChange={(e) => handleDataEntry(e.target.value, 652, 1)}
+                        />
+                        <input 
+                            type="number" 
+                            className="bin form-control input-value" 
+                            name={`${disaggregationData["653"].disaggregation_type}-${disaggregationData["653"].labels[1]}`}
+                            id={`id_${disaggregationData["653"].disaggregation_type}-${disaggregationData["653"].labels[1]}`}
+                            value={disaggregationData["653"].labels[1].value || ""}
+                            onChange={(e) => handleDataEntry(e.target.value, 653, 1)}
+                        />
+                    </div>
+                </li>
+
+                <li className="list-group-item reference-row">
+                    <div className="item__label">{gettext("Actual Total value")}</div>
+                    <div className="item__value--container">
+                        <div className="bin">{parseInt(disaggregationData['652'].labels[0].value || 0) + parseInt(disaggregationData['652'].labels[1].value || 0)}</div>
+                        <div className="bin">{parseInt(disaggregationData['653'].labels[0].value || 0) + parseInt(disaggregationData['653'].labels[1].value || 0)}</div>
+                    </div>
+                </li>
+                {
+                    formErrors.totals_error &&
+                        <span id={`validation_id_totals_error`} className="has-error">{formErrors.totals_error}</span>
+                }
+            </ul>
+        </fieldset>
+    )
+}
+
+
+
+// ***** Dissaggreagation Fields Section *****
+const DissaggregationFields = ({ indicatorID, disagg, disaggregationData, setDisaggregationData, total, title, helptext, formErrors}) => {
+    const [expanded, setExpanded] = useState(false);
+
+    let disaggID = disagg.reduce((id, disaggregation) => {
+        id = id + (id === "" ? "" : "-") + disaggregation.pk;
+        return id;
+    },"")
+
+    useEffect(() => {
+        $(`#${indicatorID}-${disaggID}`).on('show.bs.collapse', function() {
+            setExpanded(true);
+        })
+        $(`#${indicatorID}-${disaggID}`).on('hide.bs.collapse', function() {
+            setExpanded(false);
+        })
+    }, [])
+
+    let handleDataEntry = (value, inputDisaggPk, customsort) => {
+        let update = {...disaggregationData};
+        update[inputDisaggPk].labels[customsort - 1] = {...disaggregationData[inputDisaggPk].labels[customsort - 1], value: value};
+        setDisaggregationData(update);
+    }
+
+    return (
+        <fieldset>
```

Comment:
Might be useful to have comments about which sections of the disaggregation input each one of these major divs represents.